src/index.ts

```diff
@@ -74,12 +101,25 @@ async function retryFetch(
 
   try {
     const response = await promiseRetry(async (throwRetryError, retryCount) => {
+      jsonFetchOptions.onRequestStart?.({url: requestUrl, retryCount, ...jsonFetchOptions});
```

Comment:
I'm just realizing now re-reading the code, everywhere you're using `jsonFetchOptions`, I believe we should be using `requestOptions`. We want the "actual", normalized, defaulted options to be logged – I think that's less surprising and more useful.

(You'll need to change the types to reference `RequestInit` instead of `JsonFetchOptions`)

Reply:
After consider and try this suggestion, I'm not sure if that will work due that `retryFetch` is receiving `JsonFetchOptions`, or better said, will work but I think will be too verbose due that instead of spread the `JsonFetchOptions` received by **retryFetch** like this
```js
jsonFetchOptions.onRequestStart?.({url: requestUrl, retryCount, ...jsonFetchOptions});
```
will be required to destructure whatever options we want to use from  `RequestInit` instead, like this
```js
jsonFetchOptions.onRequestStart?.({
        url: requestUrl,
        retryCount,
        method: jsonFetchOptions.method,
        ...
        anyOther: jsonFetchOptions.anyOther
});
```
all this in order to satisfy to TS.

Am I missing something? What do you think?

Reply:
I got you now, sorry for the confusion, makes sense, changes applied.